rfcs/0122-system-permission.md

```diff
@@ -0,0 +1,712 @@
+| `api/users/{id}`  | `PUT`  | `Write`  | _**Users can update limited fields on their own records. Only `Project` principals can update other principals. Only `System` principals can update other principals with the `System` permission.**_ |  |
+| `api/users/{id}/searches`  | `GET`  | `Read`  | Only the principal's own search history can be retrieved. |  |
+| `api/users/{id}/searches/update`  | `POST`  | `Write`  | Only the principal's own search history can be updated. |  |
+| `api/users/{id}/unlinkauthenticationprovider`  | `POST`  | _**`System`**_  |  |  |
+| `api/users/current`  | `GET`  | `Public`  | Can only return logged-in user information. |  |
+| `api/users/login`  | `POST`  | `Public`  |  |  |
+| `api/users/logout`  | `POST`  | `Public`  |  |  |
+| `api/users/providers`  | `GET`  | `Public`  |  |  |
+| `api/users/resources`  | `GET`  | `Public`  |  |  |
+| `api/users/template`  | `GET`  | _**`Project`**_  |  |  |
+
+### `api/workspaces` 
+
+| Path | HTTP method | Permission demand | Additional requirements | Notes |
+| --- | --- | --- | --- | --- |
+| `api/workspaces/`  | `GET`  | `Read`  | Users can only read shared workspaces, and those that they own. |  |
+| `api/workspaces/`  | `POST`  | `Write`  | _**Users can only create shared workspaces or personally owned ones. The additional `Project` permission is required to create protected workspaces.**_ |  |
+| `api/workspaces/{id}`  | `DELETE`  | `Write`  | _**Users can only remove shared workspaces and those that they own. The additional `Project` permission is required to remove protected workspaces.**_ |  |
+| `api/workspaces/{id}`  | `GET`  | `Read`  | Users can only read shared workspaces, and those that they own. |  |
+| `api/workspaces/{id}`  | `PUT`  | `Write`  | _**Users can only update shared workspaces and those that they own. The additional `Project` permission is required to update protected workspaces.**_ |  |
+| `api/workspaces/resources`  | `GET`  | `Public`  |  |  |
+| `api/workspaces/template`  | `GET`  | `Write`  |  |  |
+
+### `health/` 
+
+| Path | HTTP method | Permission demand | Additional requirements | Notes |
+| --- | --- | --- | --- | --- |
+| `health/`  | `GET`  | `Public`  |  |  |
+
+### `integrated/` 
+
+| Path | HTTP method | Permission demand | Additional requirements | Notes |
+| --- | --- | --- | --- | --- |
+| `integrated/`  | `GET`  | `Public`  | The Windows identity must already be authenticated. |  |
+
+### `oidc/challenge` 
+
+| Path | HTTP method | Permission demand | Additional requirements | Notes |
+| --- | --- | --- | --- | --- |
+| `oidc/challenge`  | `GET`  | `Public`  |  |  |
+| `oidc/challenge`  | `POST`  | `Public`  |  |  |
+
+### `theme/styles.css` 
+
+| Path | HTTP method | Permission demand | Additional requirements | Notes |
+| --- | --- | --- | --- | --- |
+| `theme/styles.css`  | `GET`  | `Public`  |  |  |
+
+## Backwards Compatibility
```

Comment:
Splitting this into "Breaking Changes", "Migration", and "Backwards Compatibility" might make this section clearer?